From dbf607e74536d71063dbcecae3f50da583aa0cee Mon Sep 17 00:00:00 2001
From: Anson Huang <Anson.Huang@nxp.com>
Date: Wed, 13 Jan 2016 22:36:26 +0800
Subject: [PATCH 0845/5242] MLK-12136-2 ARM: imx: add ddr phy registers
 restore for i.MX7D

commit  33d62e426a592988b684365ce9f1c450bf3d27b0 from
https://source.codeaurora.org/external/imx/linux-imx.git

i.MX7D TO1.1 adds some DDR PHY register settings to fix the CKE
timing issue, when fast MIX off in DSM, need to restore them
to make sure the DDR PHY setting is correct.

Signed-off-by: Anson Huang <Anson.Huang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/mach-imx/pm-imx7.c |   10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/arch/arm/mach-imx/pm-imx7.c b/arch/arm/mach-imx/pm-imx7.c
index df366e7..2bc5395 100644
--- a/arch/arm/mach-imx/pm-imx7.c
+++ b/arch/arm/mach-imx/pm-imx7.c
@@ -230,6 +230,11 @@ struct imx7_pm_socdata {
 	{ 0x10, READ_DATA_FROM_HARDWARE },
 	{ 0x1c, READ_DATA_FROM_HARDWARE },
 	{ 0x9c, READ_DATA_FROM_HARDWARE },
+	{ 0x7c, READ_DATA_FROM_HARDWARE },
+	{ 0x80, READ_DATA_FROM_HARDWARE },
+	{ 0x84, READ_DATA_FROM_HARDWARE },
+	{ 0x88, READ_DATA_FROM_HARDWARE },
+	{ 0x6c, READ_DATA_FROM_HARDWARE },
 	{ 0x20, READ_DATA_FROM_HARDWARE },
 	{ 0x30, READ_DATA_FROM_HARDWARE },
 	{ 0x50, 0x01000008 },
@@ -277,6 +282,11 @@ struct imx7_pm_socdata {
 	{ 0x4, READ_DATA_FROM_HARDWARE },
 	{ 0x10, READ_DATA_FROM_HARDWARE },
 	{ 0x9c, READ_DATA_FROM_HARDWARE },
+	{ 0x7c, READ_DATA_FROM_HARDWARE },
+	{ 0x80, READ_DATA_FROM_HARDWARE },
+	{ 0x84, READ_DATA_FROM_HARDWARE },
+	{ 0x88, READ_DATA_FROM_HARDWARE },
+	{ 0x6c, READ_DATA_FROM_HARDWARE },
 	{ 0x20, READ_DATA_FROM_HARDWARE },
 	{ 0x30, READ_DATA_FROM_HARDWARE },
 	{ 0x50, 0x01000010 },
-- 
1.7.9.5

